api/agent/drivers/docker/docker_client.go

```diff
@@ -36,6 +36,7 @@ type dockerClient interface {
 	RemoveContainer(opts docker.RemoveContainerOptions) error
 	PullImage(opts docker.PullImageOptions, auth docker.AuthConfiguration) error
 	InspectImage(ctx context.Context, name string) (*docker.Image, error)
+	InspectContainerWithContext(container string, ctx context.Context) (*docker.Container, error)
```

Comment:
nit/curious: why are the method signatures on this interface like this?  Why isn't it just `InspectContainer(ctx context.Context, ...)`?

Reply:
we're just wrapping the docker driver client we use [to add timeouts / retries / some error filtering] https://godoc.org/github.com/fsouza/go-dockerclient so wanted to keep signatures consistent, fsouza definitely tacked on the 'WithContext' methods -- there is a bug filed to switch to the real docker client bindings but it will take a bit.